sim/team-validator.ts

```diff
@@ -372,12 +372,18 @@ export class TeamValidator {
 
 		let learnsetTemplate = template;
 		let tierTemplate = template;
-		if (ability.id === 'battlebond' && template.id === 'greninja' && ruleTable.has('obtainableformes')) {
-			tierTemplate = learnsetTemplate = dex.getTemplate('greninjaash');
-			if (set.gender && set.gender !== 'M') {
-				problems.push(`Battle Bond Greninja must be male.`);
+		if (ability.id === 'battlebond' && template.id === 'greninja') {
+			let ashTemplate = dex.getTemplate('greninjaash');
+			if (ruleTable.has('obtainableformes')) {
+				tierTemplate = ashTemplate;
+			}
+			learnsetTemplate = ashTemplate;
```

Comment:
This seems like an unnecessarily complicated way to write

```ts
			learnsetTemplate = dex.getTemplate('greninjaash');
			if (ruleTable.has('obtainableformes')) {
				tierTemplate = learnsetTemplate;
			}
```

Reply:
I think I was considering putting the learnset assignment behind another condition like if (ruleTable.has('obtainablemoves')) but it seemed redundant, so it ended up like that. What you wrote seems like it should be slightly more performative for what it's doing now, so I updated it to that.